### ISSUE TITLE
More closely emulate profiles in `~/.aws/config` for AWS_DEFAULT_REGION

### ISSUE THREAD

**Issue by synfinatic** [synfinatic/aws-sso-cli#152]
So the overall goal should be to as closely as possible emulate AWS CLI profile feature.

Need two changes:
 1. Remove `--region` flag.  Users should always just use the `$AWS_DEFAULT_REGION` to be consistent
 2. Ensure that when the user has defined a `DefaultRegion` in the config file, it does not overwrite a user-defined `$AWS_DEFAULT_REGION` value.  <== (`$__AWS_SSO_DEFAULT_REGION` to track???)

What is unclear:
 1. User calls `eval/exec` -> sets `$AWS_DEFAULT_REGION` because it is not set in the shell, but they have a `DefaultRegion`
 2. User calls `eval` again, and a different DefaultRegion is set.  We need to know to overwrite the previous value?  This would most closely emulate the way profiles work in `~/.aws/config`
 3. When user calls `eval --clean` we need to know to delete the `$AWS_DEFAULT_REGION` since we set that??

Possible solution: a config file feature which never/always clears the variable?  Or a `--clear-all` vs `--clear` ?  This way the user can be more explicit and doesn't have to remember.  If we are skipping overwriting the variable, we can always generate a warning.

Possible solution 2: create a mirror variable `$__AWS_SSO_DEFAULT_REGION` which is set to whatever we set to `$AWS_DEFAULT_REGION` and if the two are the same, then you can clear/overwrite it.  If the values are different (because user changed it, we should clear the tracking variable.

More info see commentary in: #149 

Basically:

```bash
if [ -z "$ADR" -o "$ADR" = "$ASDR"]; then
   # only set AWS_DEFAULT_REGION if it is not already set or we are managing it
   # these could also be "unset" commands if user is calling `eval --clear`
   export ADR=X
   export ASDR=X
else  
   # clear our flag so we don't get confused into thinking 
   # we are managing it if user sets ADR to our value
   unset ASDR
fi
```

Also, need to add `--no-region` to allow users the option of not using the `DefaultRegion`
